phoenix-core/src/main/java/org/apache/phoenix/mapreduce/index/IndexScrutinyMapper.java

```diff
@@ -254,6 +291,47 @@ private void processBatch(Context context)
         }
     }
 
+    private void checkIfInvalidRowsExpired(Context context,
+            Map<String, Pair<Long,
+            List<Object>>> targetPkToSourceValues) throws SQLException, IOException {
+        long ttl = getTableTtl();
+        Set<Map.Entry<String, Pair<Long,
+                                List<Object>>>> entrySet = targetPkToSourceValues.entrySet();
+
+        Iterator<Map.Entry<String, Pair<Long,
+                        List<Object>>>> itr = entrySet.iterator();
```

Comment:
nit: Splitting of these templatized types across multiple lines is making it worse than long lines. Missing typedefs in Java :|